.buildkite/pipeline.yml

```diff
@@ -128,3 +135,100 @@ steps:
       queue: mac
     env:
       IMAGE_ID: xcode-14.3.1
+
+  - label: iOS Build and SauceLabs
+    key: ios-build-and-saucelabs
+    command: .buildkite/build-ios.sh
+    plugins:
+      - automattic/a8c-ci-toolkit#2.18.2
+      - *git-cache-plugin
+    artifact_paths:
+      - ./gutenberg/packages/react-native-editor/ios/GutenbergDemo.app.zip
+    agents:
+      queue: mac
+    env: *xcode_agent_env
+
+  - label: Test iOS on Device – Canary Pages
+    depends_on: ios-build-and-saucelabs
+    command: .buildkite/test-ios.sh --canary
+    plugins:
+      - automattic/a8c-ci-toolkit#2.18.2
+      - *git-cache-plugin
+    artifact_paths:
+      - reports/test-results/ios-test-results.xml
+    agents:
+      queue: mac
+    env: *xcode_agent_env
+    notify:
+      - github_commit_status:
+          context: Test iOS on Device - Canaries
+
```

Comment:
There is quite a bit of duplication below because of the different steps requirement depending on the branch we're in. That is, on `trunk`, `release/` and Dependabot branches, we run _all_ UI test suites, while on other branches we want a block step to manually trigger the full UI test suites.

In the future, I think we should consider using a dynamic pipeline to avoid duplication and make changes simpler. But it seems out of scope for this PR.